conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -0,0 +1,346 @@
+        if (HttpHeaders.CONTENT_TYPE.equalsIgnoreCase(headerName)) {
+            return false;
+        }
+        if (HttpHeaders.CONTENT_LENGTH.equalsIgnoreCase(headerName)) {
+            return false;
+        }
+        // Tracing path template is informational only
+        if (PATH_TEMPLATE.equalsIgnoreCase(headerName)) {
+            return false;
+        }
+        return true;
+    }
+
+    private static String urlDecode(String input) {
+        if (input.contains("%")) {
+            try {
+                return URLDecoder.decode(input, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                throw new SafeRuntimeException("Failed to decode path segment", e);
+            }
+        }
+        return input;
+    }
+
+    private static Optional<RequestBody> requestBody(Request request) {
+        byte[] requestBodyContent = request.body();
+        if (requestBodyContent == null) {
+            return Optional.empty();
+        }
+        Optional<String> maybeContentType = getFirstHeader(request, HttpHeaders.CONTENT_TYPE);
+        if (!maybeContentType.isPresent()) {
+            if (requestBodyContent.length == 0) {
+                return Optional.empty();
+            }
+            throw new SafeIllegalStateException("A Content-Type header was not present on feign request with a body");
+        }
+        return Optional.of(new ByteArrayRequestBody(requestBodyContent, maybeContentType.get()));
+    }
+
+    private static Optional<String> getFirstHeader(Request request, String name) {
+        Collection<String> values = request.headers().get(name);
+        if (values != null) {
+            return Optional.ofNullable(Iterables.getFirst(values, null));
+        }
+        return Optional.empty();
+    }
+
+    private static final class ByteArrayRequestBody implements RequestBody {
+
+        private final byte[] buffer;
+        private final String contentType;
+
+        ByteArrayRequestBody(byte[] buffer, String contentType) {
+            this.buffer = buffer;
+            this.contentType = contentType;
+        }
+
+        @Override
+        public void writeTo(OutputStream output) throws IOException {
+            output.write(buffer);
+        }
+
+        @Override
+        public String contentType() {
+            return contentType;
+        }
+
+        @Override
+        public boolean repeatable() {
+            return true;
+        }
+
+        @Override
+        public void close() {
+            // nothing to do
+        }
+    }
+
+    private static final class DialogueResponseBody implements feign.Response.Body {
+
+        private final Response response;
+
+        DialogueResponseBody(Response response) {
+            this.response = response;
+        }
+
+        @Override
+        public Integer length() {
+            return response.getFirstHeader(HttpHeaders.CONTENT_LENGTH)
+                    .map(Ints::tryParse)
```

Comment:
interestingly the implementation in OkhttpClient also verifies this is > 0, which presumably allows clients to keep working (and allocating properly) even if some bizarre server returns content-length: -1